src/BuiltInTools/dotnet-watch/dotnet-watch.csproj

```diff
@@ -23,6 +23,13 @@
     <PackageReference Include="Microsoft.Build.Locator" Version="$(MicrosoftBuildLocatorPackageVersion)" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Features" Version="$(MicrosoftCodeAnalysisCSharpFeaturesPackageVersion)" />
     <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.MSBuild" Version="$(MicrosoftCodeAnalysisWorkspacesMSBuildPackageVersion)" />
+
+    <!--
+      Explicit System.Text.Json package reference is required for source-build to pick up the live package.
+      Avoids picking up an old version via transitive dependency from Microsoft.Build or Microsoft.CodeAnalysis.Workspaces.MSBuild.
+    -->
+    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonVersion)" Condition="'$(DotnetBuildFromSource)' == 'true'" />
```

Comment:
Same question as your other PR.  Is the source-build condition necessary?  This change will auto-merge up to main so this one feels worthwhile to do the right action in servicing.

Reply:
I see - since it auto-flows to `main` it makes sense to remove the condition - will update this.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/33618/commits/1a814df37d3cf3a894d1af060d6073fe1d595f22

I have not modified the comment as it adds the value.